bambou/session.go

```diff
@@ -121,14 +130,25 @@ func (s *Session) makeAuthorizationHeaders() (string, *Error) {
 
 func (s *Session) prepareHeaders(request *http.Request, info *FetchingInfo) *Error {
 
-	authString, berr := s.makeAuthorizationHeaders()
-	if berr != nil {
-		return berr
+	if s.Certificate != nil { // We're using X509 certificate based auth.
+
+		// XXX - "InsecureSkipVerify"
+		s.client.Transport = &http.Transport{TLSClientConfig: &tls.Config{Certificates: []tls.Certificate{*s.Certificate}, InsecureSkipVerify: true}}
+
+	} else { // We're using user & password based authentication
+
+		// Skip TLS certificate verification
+		s.client.Transport = &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
```

Comment:
As in comment previously: shouldn't forcing TLS verification to be disabled a user choice?